### ISSUE TITLE
Issues with TVars parsing

### ISSUE THREAD

**Issue by GeorgeR227** [AlgebraicJulia/Decapodes.jl#84]
I was trying to make a decapode where nothing else was pointing into the TVar variable Ḣ but I ran into an issue where for some reason, both the Ḣ I declared and an auto-generated Ḣ, which is the actual TVar here, were created.
```Julia
DecaTest10 = quote
  A::Form0{X}
  B::Form0{X}
  C::Form0{X}
  Ḣ::Form5{X}
  H::Form5{X}

  A == b(B)
  A == c(C)
  ∂ₜ(H) == Ḣ
end

Test10 = SummationDecapode(parse_decapode(DecaTest10))

SummationDecapode{Any, Any, Symbol} with elements Var = 1:6, TVar = 1:1, Op1 = 1:3, Op2 = 1:0, Σ = 1:0, Summand = 1:0
┌─────┬───────┬──────┐
│ Var │  type │ name │
├─────┼───────┼──────┤
│   1 │ Form0 │    A │
│   2 │ Form0 │    B │
│   3 │ Form0 │    C │
│   4 │ Form5 │    Ḣ │
│   5 │ Form5 │    H │
│   6 │ infer │    Ḣ │
└─────┴───────┴──────┘
┌──────┬──────┐
│ TVar │ incl │
├──────┼──────┤
│    1 │    6 │
└──────┴──────┘
┌─────┬─────┬─────┬─────┐
│ Op1 │ src │ tgt │ op1 │
├─────┼─────┼─────┼─────┤
│   1 │   2 │   1 │   b │
│   2 │   3 │   1 │   c │
│   3 │   5 │   6 │  ∂ₜ │
└─────┴─────┴─────┴─────┘
```
![TVar1](https://user-images.githubusercontent.com/78235421/201766436-0f15407c-ef6b-4cbe-b403-d4bc7a462256.svg)

So I decided to delete my Ḣ since it wasn't being used, but now it won't compile since I didn't declare Ḣ, even though it's not being used.
```Julia
DecaTest10 = quote
  A::Form0{X}
  B::Form0{X}
  C::Form0{X}
  H::Form5{X}

  A == b(B)
  A == c(C)
  ∂ₜ(H) == Ḣ
end

Test10 = SummationDecapode(parse_decapode(DecaTest10))

KeyError: key :Ḣ not found
Stacktrace:
 [1] getindex(h::Dict{Symbol, Int64}, key::Symbol)
   @ Base .\dict.jl:498
 [2] reduce_term!(t::Decapodes.Var, d::SummationDecapode{Any, Any, Symbol}, syms::Dict{Symbol, Int64})
   @ Decapodes c:\Users\19542\Documents\GitHub\Decapodes.jl\src\Decapodes2\language.jl:69
 [3] eval_eq!(eq::Eq, d::SummationDecapode{Any, Any, Symbol}, syms::Dict{Symbol, Int64})
   @ Decapodes c:\Users\19542\Documents\GitHub\Decapodes.jl\src\Decapodes2\language.jl:114
 [4] SummationDecapode(e::Decapodes.DecaExpr)
   @ Decapodes c:\Users\19542\Documents\GitHub\Decapodes.jl\src\Decapodes2\language.jl:186
 [5] top-level scope
   @ c:\Users\19542\Documents\GitHub\Decapodes.jl\test\Decapodes2\rewrite.jl:526
```
I suspect that this occurs because we assumed that "floating" variables without edges pointing into them, like Ḣ, would have exactly one operation leading into them.

**Comment by lukem12345** [comment]
The following example helps explain why auto-generating a `Var` like Ḣ is sometimes necessary. These two Decapodes should encode the same physics:

```julia
Exp = quote
  A::Form0{X}
  H::Form0{X}
  Ḣ::Form0{X}

  ∂ₜ(H) == Ḣ
  Ḣ == f(A)
end
```
```julia
Imp = quote
  A::Form0{X}
  H::Form0{X}

  ∂ₜ(H) == f(A)
end
```
That is, the update variable for H is explicit in `Exp` and implicit in `Imp`. When we create a Decapode, there must be an intermediary variable that we can set both sides of this equation equal to.

When we solve this issue, we can either make the parsing code more sophisticated, or make the pre-processing of a Decapode more sophisticated. We prefer this second option whenever possible for a few reasons. We want to be able to use as much declarative-style Catlab machinery as possible, and writing bug-free parsing code is tough.

**Comment by lukem12345** [comment]
Closed by #95
